pyproject.toml

```
@@ -1,3 +1,177 @@
+    "Natural Language :: English",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+]
+dependencies = [
+    "overrides!=7.6.0",
+    "pydantic>=2.0.0",
+    "pyxdg",
+    "PyYAML",
+    "requests-unixsocket2>=0.4.0",
+    "requests>=2.32,<3.0",
+]
+
+[project.optional-dependencies]
+apt = [
+    "python-apt>=2.4.0;sys_platform=='linux'"
+]
+docs = [
+    "canonical-sphinx",
+    "sphinx",
+    "sphinx-autodoc-typehints",
+    "sphinx-lint",
+    "sphinx-pydantic",
+    "sphinx-rtd-theme",
+    "sphinxcontrib-details-directive",
+]
+dev = [  # TODO: Remove this once we've switched CI to uv
+    # Inherited from starbase
+    "build",
+    "coverage[toml]~=7.4",
+    "pytest~=8.0",
+    "pytest-cov~=5.0",
+    "pytest-mock~=3.12",
+    "yamllint~=1.34",
+    "mypy[reports]~=1.11.0",
+    "pyright==1.1.388",
+    "types-Pygments",
+    "types-colorama",
+    "types-setuptools",
+
+    # Project-specific linting
+    "autoflake",
+    "black",
+    "codespell",
+    "pydocstyle",
+    "tox",
+
+    # Testing
+    "hypothesis",
+    "jsonschema",
+    "pytest-check",
+    "pytest-subprocess",
+    "requests-mock",
+
+    # Type checking
+    "types-colorama",
+    "types-docutils",
+    "types-jsonschema",
+    "types-Pillow",
+    "types-Pygments",
+    "types-pytz",
+    "types-PyYAML",
+    "types-requests",
+    "types-setuptools",
+]
+
+[tool.uv]
+constraint-dependencies = [
+    # Basic constraints to allow --resolution=lowest
+    "build>=0.7.0",
+    "iniconfig>=1.1.0",
+    "lxml>=5.0",
+    "pyparsing>=3.0.0",
+    "pyproject-hooks>=1.0.0",
+    "pyyaml>=5.0",
+    "markdown>=3.0",
+    "markupsafe>=2.0",
+    "pyyaml>5.0",
+    "regex>=2021.11.10",
+    "sphinx-basic-ng>=1.0.0b1",
+    "tornado>=4.0",
+    "webencodings>=0.4.0",
+]
+dev-dependencies = [
+    # Inherited from starbase
+    "build",
+    "coverage[toml]~=7.4",
+    "pytest~=8.0",
+    "pytest-cov~=5.0",
+    "pytest-mock~=3.12",
+    "yamllint~=1.34",
+    "mypy[reports]~=1.11.0",
+    "pyright==1.1.388",
+    "types-Pygments",
+    "types-colorama",
+    "types-setuptools",
+
+    # Project-specific linting
+    "autoflake",
+    "black",
+    "codespell",
+    "pydocstyle",
+
+    # Testing
+    "hypothesis",
+    "jsonschema",
+    "pytest-check",
+    "pytest-subprocess",
+    "requests-mock",
+
+    # Type checking
+    "types-colorama",
+    "types-docutils",
+    "types-jsonschema",
+    "types-Pillow",
+    "types-Pygments",
+    "types-pytz",
+    "types-PyYAML",
+    "types-requests",
+    "types-setuptools",
+]
+
+
+[project.scripts]
+craftctl = "craft_parts.ctl:main"
+
+[project.urls]
+Homepage = "https://github.com/canonical/craft-parts"
+
+[build-system]
+requires = [
+    "setuptools>=69.0",
+    "setuptools_scm[toml]>=7.1"
+]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools.package-dir]
+"craft_parts_docs" = "docs/common"
```

when adding this it seems it disables automatic package discovery, which is the root cause to the import errors for craft parts itself

```suggestion
[tool.setuptools.package-dir]
"craft_parts" = "craft_parts"
"craft_parts_docs" = "docs/common"
```